lib/ex_aws/request.ex

```diff
@@ -9,14 +9,7 @@ defmodule ExAws.Request do
   @type error_t :: {:error, {:http_error, http_status, binary}}
   @type response_t :: success_t | error_t
 
-  def request(http_method, url, data, headers, config, service) do
-    body =
-      case data do
-        [] -> "{}"
```

Comment:
When is `[]` possible? Does it represent `nil`? Otherwise an empty list and an empty object are not equivalent in JSON.